tests/data/drivers/test_sentinel_query_reader.py

```diff
@@ -0,0 +1,214 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""Github Sentinel Query repo import class and helpers tests."""
+
+import os
+from datetime import datetime
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+import pytest_check as check
+
+from msticpy.data.drivers.sentinel_query_reader import (
+    SentinelQuery,
+    _create_queryfile_metadata,
+    _format_query_name,
+    _import_sentinel_query,
+    _organize_query_list_by_folder,
+    get_sentinel_queries_from_github,
+    import_sentinel_queries,
+    read_yaml_files,
+    write_to_yaml,
+)
+
+from ...unit_test_lib import get_test_data_path
+
+# variables used throughout tests
+DEF_PATH = Path.joinpath(Path(os.getcwd()))
+BASE_DIR = Path.joinpath(DEF_PATH, Path("Azure-Sentinel-master"))
+BASE_DIR_TEST_FOLDER = Path.joinpath(get_test_data_path(), "sentinel_query_import_data")
+_SENTINEL_QUERY_READER = "msticpy.data.drivers.sentinel_query_reader"
+
+
+@patch(f"{_SENTINEL_QUERY_READER}.get_sentinel_queries_from_github")
+def test_get_sentinel_queries_from_github(get_sentinel_queries_from_github):
+    get_sentinel_queries_from_github.return_value = True
+    assert get_sentinel_queries_from_github(outputdir=os.getcwd())
+    get_sentinel_queries_from_github.assert_called_once()
+
+
+# original test case for downloading github zip
+@pytest.mark.skip(reason="requires downloading the file directly during the test")
+def test_get_sentinel_queries_from_github():
+    assert get_sentinel_queries_from_github(outputdir=os.getcwd())
+
+
+def test_read_yaml_files():
+    yaml_files = read_yaml_files(parent_dir=BASE_DIR_TEST_FOLDER, child_dir="Detections")
+    assert yaml_files[str(BASE_DIR_TEST_FOLDER.joinpath("Detections/Anomalies/UnusualAnomaly.yaml"))]
+
+
+def test__import_sentinel_query():
```

Comment:
extra "_" in name

Reply:
There are two test functions - one for _import_sentinel_query and one for import_sentinel_query so that's why there's an extra "_"